crates/bevy_ecs/src/query/state.rs

```diff
@@ -74,6 +74,175 @@ pub struct QueryState<D: QueryData, F: QueryFilter = ()> {
     par_iter_span: Span,
 }
 
+/// View into a [`QueryState`] to avoid split-borrow issues when using [`QueryState`]
```

Comment:
Maybe it's nice to also add the link to the babysteps article. You'd have to check in with other maintainers about what the policy on adding links is because link rot is a genuine concern, and maybe it's enough to have it in the PR.

My general impression of this trick is that it's very clever, but adds a layer of obfuscation that makes the code more difficult to understand. However, it does work / compiles, so it's good enough.

Reply:
I am a bit wary about adding external links like this, which is why I simply linked the article in the PR for now. But that's something to discuss, yes